cdap-app-templates/cdap-etl/cdap-etl-tools/src/main/java/co/cask/cdap/etl/tool/UpgradeTool.java

```diff
@@ -0,0 +1,284 @@
+      Integer.parseInt(commandLine.getOptionValue("t")) : DEFAULT_READ_TIMEOUT;
+    ClientConfig.Builder clientConfigBuilder = ClientConfig.builder()
+      .setDefaultReadTimeout(readTimeout)
+      .setConnectionConfig(connectionConfig);
+
+    if (commandLine.hasOption("a")) {
+      String tokenFilePath = commandLine.getOptionValue("a");
+      File tokenFile = new File(tokenFilePath);
+      if (!tokenFile.exists()) {
+        throw new IllegalArgumentException("Access token file " + tokenFilePath + " does not exist.");
+      }
+      if (!tokenFile.isFile()) {
+        throw new IllegalArgumentException("Access token file " + tokenFilePath + " is not a file.");
+      }
+      String tokenValue = new String(Files.readAllBytes(tokenFile.toPath()), StandardCharsets.UTF_8).trim();
+      AccessToken accessToken = new AccessToken(tokenValue, 82000L, "Bearer");
+      clientConfigBuilder.setAccessToken(accessToken);
+    }
+
+    return clientConfigBuilder.build();
+  }
+
+  private static void convertFile(String configFilePath) throws IOException {
+    File configFile = new File(configFilePath);
+    if (!configFile.exists()) {
+      throw new IllegalArgumentException(configFilePath + " does not exist.");
+    }
+    if (!configFile.isFile()) {
+      throw new IllegalArgumentException(configFilePath + " is not a file.");
+    }
+    String fileContents = new String(Files.readAllBytes(configFile.toPath()), StandardCharsets.UTF_8);
+
+    ETLAppRequest artifactFile = GSON.fromJson(fileContents, ETLAppRequest.class);
+    if (!shouldUpgrade(artifactFile.artifact)) {
+      throw new IllegalArgumentException(
+        "Cannot update for artifact " + artifactFile.artifact + ". " +
+          "Please check the artifact is cdap-etl-batch or cdap-etl-realtime in the system scope of version 3.2.x.");
+    }
+
+    String version = getVersion();
+
+    if (BATCH_NAME.equals(artifactFile.artifact.getName())) {
+      ArtifactSummary artifact = new ArtifactSummary(BATCH_NAME, version, ArtifactScope.SYSTEM);
+      OldETLBatchConfig oldConfig = GSON.fromJson(artifactFile.config, OldETLBatchConfig.class);
+      AppRequest<ETLBatchConfig> updated = new AppRequest<>(artifact, oldConfig.getNewConfig());
+      System.out.println(GSON.toJson(updated));
+    } else {
+      ArtifactSummary artifact = new ArtifactSummary(REALTIME_NAME, version, ArtifactScope.SYSTEM);
+      OldETLRealtimeConfig oldConfig = GSON.fromJson(artifactFile.config, OldETLRealtimeConfig.class);
+      AppRequest<ETLRealtimeConfig> updated = new AppRequest<>(artifact, oldConfig.getNewConfig());
+      System.out.println(GSON.toJson(updated));
+    }
+  }
+
+  // reads current version from the etl.properties file contained in resources.
+  private static String getVersion() {
+    Properties prop = new Properties();
+    try {
+      URL resource = UpgradeTool.class.getResource("/etl.properties");
+      try (InputStream in = UpgradeTool.class.getResourceAsStream("/etl.properties")) {
```

Comment:
how would the user set the version ? looks like this file is bundled in this jar..